koku/api/report/queries.py

```diff
@@ -608,8 +628,16 @@ def execute_query(self):
                 query_order_by = query_order_by + ('rank',)
 
             query_data = query_data.order_by(*query_order_by)
-            data = self._apply_group_by(list(query_data))
-            data = self._transform_data(query_group_by, 0, data)
+            is_csv_output = self._accept_type and 'text/csv' in self._accept_type
```

Comment:
This seems weird to me, but that could be because I don't fully understand where this sits in the stack.

I would not expect this class to care about the final data serialization method. Shouldn't this be returning data to the serializer, then just let the serializer handle the final rendering into JSON or CSV or whatever?

The main question I have is whether this is going to be maintainable as we add other possible operations or MIME-types?  It feels like this could get really complex really quickly.

Reply:
Our current csv output doesn’t look great because we are transforming the data. This essentially says skip that step because csv doesn’t want that. Theoretically we could move the transformation to a JSON renderer layer or have a use case to return “flattened” JSON and CSV & the query param would do the same thing. Just not on the requirements list yet.